crates/payload/primitives/src/traits.rs

```diff
@@ -21,35 +17,26 @@ pub trait PayloadBuilder: Send + Unpin {
     /// The error type returned by the builder.
     type Error;
 
-    /// Sends a message to the service to start building a new payload for the given payload
-    /// attributes and returns a future that resolves to the payload.
-    async fn send_and_resolve_payload(
+    /// Sends a message to the service to start building a new payload for the given payload.
+    ///
+    /// Returns a receiver that will receive the payload id.
+    fn send_new_payload(
         &self,
         attr: <Self::PayloadType as PayloadTypes>::PayloadBuilderAttributes,
-    ) -> Result<PayloadFuture<<Self::PayloadType as PayloadTypes>::BuiltPayload>, Self::Error>;
+    ) -> oneshot::Receiver<Result<PayloadId, Self::Error>>;
 
     /// Returns the best payload for the given identifier.
     async fn best_payload(
         &self,
         id: PayloadId,
     ) -> Option<Result<<Self::PayloadType as PayloadTypes>::BuiltPayload, Self::Error>>;
 
-    /// Sends a message to the service to start building a new payload for the given payload.
-    ///
-    /// This is the same as [`PayloadBuilder::new_payload`] but does not wait for the result
-    /// and returns the receiver instead
-    fn send_new_payload(
-        &self,
-        attr: <Self::PayloadType as PayloadTypes>::PayloadBuilderAttributes,
-    ) -> oneshot::Receiver<Result<PayloadId, Self::Error>>;
-
-    /// Starts building a new payload for the given payload attributes.
-    ///
-    /// Returns the identifier of the payload.
-    async fn new_payload(
+    /// Resolves the payload job and returns the best payload that has been built so far.
+    async fn resolve(
         &self,
-        attr: <Self::PayloadType as PayloadTypes>::PayloadBuilderAttributes,
-    ) -> Result<PayloadId, Self::Error>;
+        id: PayloadId,
+        wait_for_pending: bool,
+    ) -> Option<Result<<Self::PayloadType as PayloadTypes>::BuiltPayload, Self::Error>>;
```

Comment:
hmm, I'd like to keep this behaviour as is but add this functionality as an additional function instead, in general not a fan of boolean args for things like this